### PR TITLE
[RNMobile] Display custom color value in mobile Cover Block color picker

### PR DESCRIPTION
**Pull request by derekblank**
## What?
Displays the hex color value when selecting a custom color on the mobile editor. Fixes:

* https://github.com/WordPress/gutenberg/issues/51283

Gutenberg Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/5852

## Why?
For color contrast accessibility reasons, it is helpful to have the value of the color displayed in a text or number format other than the color itself. Also, displaying the hex value matches other Gutenberg color picker implementations on web and mobile (like the paragraph block, for example). 

## How?
When a custom color is selected in the Cover block, replace the "Select a color" label with the hex color value within the bottom sheet.

## Testing Instructions
1. Create a post on mobile
2. Create a Cover block
3. Move the selector to select a custom color
4. Note that the BottomSheet label changes from "Select a color" to the hex color value

## Screenshots or screencast 
(Note: the video player controls may obscure the actual PR changes at bottom of this video. Ensure you are not hovering over the video in order to see the hex value.)

Before | After
-|-
<video src="https://github.com/WordPress/gutenberg/assets/643285/c79150bd-2d3a-422c-bb83-c14a8f28547e" /> | <video src="https://github.com/WordPress/gutenberg/assets/643285/d76e840a-d0a3-4691-9912-fb30bb63cfc3" />

### CODE REVIEW

File: packages/block-library/src/cover/edit.native.js [hunk 1]
```diff
@@ -364,6 +365,21 @@ const Cover = ( {
 		} );
 	}, [] );
 
+	const selectedColorText = getStylesFromColorScheme(
+		styles.selectedColorText,
+		styles.selectedColorTextDark
+	);
+
+	function getBottomLabelText() {
```
Comment (geriux) [inline packages/block-library/src/cover/edit.native.js]: I was wondering if we could have a variable instead of a function, what do you think?

**Comment by geriux** [comment]
I've tested these changes and they're working well on both iOS and Android 🎉 

Can we add an integration test for this new functionality?

**Comment by derekblank** [comment]
@dcalhoun If you have a moment to take a look, I'm curious if you have an insight into why [this integration test](https://github.com/WordPress/gutenberg/pull/51414/files#diff-dfae1f13683487e921b48dbd31985e95506e1b717eb531034d271d2e47988619R534-R568) for this component might be failing. When running the test for `packages/block-library/src/cover/test/edit.native.js`, I receive this error:

```
  ● color settings › displays the hex color value in the custom color picker

    TypeError: Cannot read properties of undefined (reading 'paddingLeft')

      49 |
      50 |      const {
    > 51 |              paddingLeft: spacing,
         |              ^
      52 |              height: pickerHeight,
      53 |              borderRadius,
      54 |      } = styles.picker;
```

I would expect that this error is due to this stylesheet object not being mocked somewhere. I am not too familiar with mocking style objects like this, but noted [this code block](https://github.com/WordPress/gutenberg/blob/04c50753d4a44b7bcc5d448517839bec73b72a18/packages/block-library/src/cover/test/edit.native.js#L34-L44) in the same test referencing mocked stylesheet objects:

https://github.com/WordPress/gutenberg/blob/04c50753d4a44b7bcc5d448517839bec73b72a18/packages/block-library/src/cover/test/edit.native.js#L34-L44

Could they be related, perhaps?

**Comment by dcalhoun** [comment]
> I would expect that this error is due to this stylesheet object not being mocked somewhere. I am not too familiar with mocking style objects like this

@derekblank I opened https://github.com/WordPress/gutenberg/pull/52943 to fix the style failures and others identified when proceeding further. That PR includes inline comments explaining the fixes as well. Please feel free to merge, cherry pick, or disregard the commits. Hope this helps.

**Review by geriux — APPROVED** [review]
Nice work! 🚀

**Pull request merged**